rosetta/transactor/transactor.go

```diff
@@ -0,0 +1,592 @@
+// Copyright 2021 Optakt Labs OÜ
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+package transactor
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"sort"
+	"strconv"
+
+	"github.com/onflow/cadence"
+	cjson "github.com/onflow/cadence/encoding/json"
+	"github.com/onflow/flow-go-sdk/crypto"
+
+	sdk "github.com/onflow/flow-go-sdk"
+	"github.com/onflow/flow-go/model/flow"
+
+	"github.com/optakt/flow-dps/models/dps"
+	"github.com/optakt/flow-dps/rosetta/failure"
+	"github.com/optakt/flow-dps/rosetta/identifier"
+	"github.com/optakt/flow-dps/rosetta/object"
+)
+
+const (
+	authorizersRequired = 1 // we only support one authorizer per transaction
+	argsRequired        = 2 // transactions need to arguments (amount & receiver)
+)
+
+// Transactor has the capabilities to determine transaction intent from an array
+// of Rosetta operations, create a Flow transaction from a transaction intent
+// and transate a Flow transaction back to an array of Rosetta operations.
+type Transactor struct {
+	validate Validator
+	generate Generator
+	invoke   Invoker
+}
```

Comment:
Don't forget to add unit tests for this component.

Reply:
Tests will be added for the Construction API after the merge, as there is a certain time critical component to the work, and at least some relevant testing was done in an exploratory manner using the Rosetta CLI compliance checks.